### PR TITLE
Use catch instead of gtest

### PR DESCRIPTION
**Pull request by drbenmorgan**
Migrate template test to use Catch instead of GTest. Latter requires a
more complex vendoring as Google recommend it be vendored in projects to
ensure compiler/binary compatibility. Catch is a single header file
and provides all needed functionality for basic projects.

Use BUILD_TESTING option as supplied by inclusion of CTest to make
recursion into `tests` subdirectory optional. It is ON by default,
so tests are always built unless explicitly requested.

Import latest catch.hpp from

https://github.com/philsquared/Catch

and place in dedicated subdirectory to ensure explicit include path.
Create INTERFACE library target for header, allowing clients simply
to link to it and pick up all requisite build settings. Catch is
Boost licensed, so compatible with current HSF licenses.

Update unit test of Example class to demo Catch based unit test program.
Document links to upstream catch tutotial and reference manual.

### CODE REVIEW

File: project_template/package/tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+target_include_directories(catch INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/catch)
+
+
+# - Create and define test executables
+# Here a generic example for creating an executable then defining it as a
+# test to CTest
+# 1. Create executable
+add_executable(test_example test_example.cpp)
+
+# 2. Link it to the librayr being tested, plus the unit test library(ies)
```
Comment (hegner) [inline project_template/package/tests/CMakeLists.txt]: librayr -> library

**Comment by hegner** [comment]
Looks nice, though slightly verbose. But we can balance that with the docs later :-)

**Comment by drbenmorgan** [comment]
O.k., typo fixed!

On the verbosity, do you mean the comments or the code? The comments can certainly be added into the docs then stripped back in the CMakeLists.txt!

**Comment by hegner** [comment]
Thanks. Can you as well resolve the conflicts w/ the master? 

And I meant the verbosity of the comments only.

**Comment by drbenmorgan** [comment]
Whoops, forgot about that, should now be good to go!

**Pull request merged**